libs/banglejs/jswrap_bangle.c

```diff
@@ -1412,6 +1423,24 @@ void peripheralPollHandler() {
     }
     if (faceUpTimer<TIMER_MAX) faceUpTimer += pollInterval;
     if (faceUpTimer>=300 && !faceUpSent) {
+      // bypass lock. wake does not consume. do not extend wake
+      if (faceUp && (bangleFlags&JSBF_WAKEON_FACEUP)) {
+        bool woke = false;
+        // LCD was turned off, turn it back on
+        if (lcdPowerTimeout && !(bangleFlags&JSBF_LCD_ON)) {
+          woke = true;
+          jswrap_banglejs_setLCDPower(1);
```

Comment:
Is there a reason you don't just do `bangleTasks |= JSBT_LCD_ON;` here like for other events? Or is it just that you moved the code and didn't change it?

The problem is you moved this code from outside of an interrupt to inside an interrupt, and it's probably not safe to call these functions from an IRQ since they could be being called from outside an IRQ at the same time.

Reply:
Well spotted.  I am going to force push with those changes, as to not spam commits.